src/app/clusters/messages-server/messages-server.cpp

```diff
@@ -182,9 +182,51 @@ bool emberAfMessagesClusterPresentMessagesRequestCallback(
     auto & responses      = commandData.responses;
 
     Delegate * delegate = GetDelegate(endpoint);
-    VerifyOrExit(isDelegateNull(delegate, endpoint) != true, err = CHIP_ERROR_INCORRECT_STATE);
+    VerifyOrExit(isDelegateNull(delegate, endpoint) != true, status = Status::NotFound);
+
+    VerifyOrExit(messageId.size() == kMessageIdLength,
+                 ChipLogProgress(Zcl, "emberAfMessagesClusterPresentMessagesRequestCallback invalid message id length");
+                 status = Status::ConstraintError);
+
+    VerifyOrExit(messageText.size() <= kMessageTextLengthMax,
+                 ChipLogProgress(Zcl, "emberAfMessagesClusterPresentMessagesRequestCallback invalid message text length");
+                 status = Status::ConstraintError);
+
+    if (responses.HasValue())
+    {
+        size_t size = 0;
+        err         = responses.Value().ComputeSize(&size);
+        VerifyOrExit(err == CHIP_NO_ERROR,
+                     ChipLogProgress(Zcl, "emberAfMessagesClusterPresentMessagesRequestCallback size check failed");
+                     status = Status::ConstraintError);
+
+        VerifyOrExit(size <= kMessageMaxOptionCount,
```

Comment:
So this needs to actually be in the spec, right?  Right now there is no constraint like this in the spec.  Spec issue needed.

Reply:
created https://github.com/CHIP-Specifications/connectedhomeip-spec/issues/8931